db/types/hintsets.py

```diff
@@ -50,8 +50,9 @@ def _add_to_db_type_hintsets(db_types, hints):
     _add_to_db_type_hintsets(categories.DATE_TYPES, (hints.date,))
     _add_to_db_type_hintsets(categories.DATETIME_TYPES, (hints.date, hints.time,))
     _add_to_db_type_hintsets(categories.DURATION_TYPES, (hints.duration,))
-    _add_to_db_type_hintsets(categories.COMPARABLE_TYPES, (hints.comparable,))
+    _add_to_db_type_hintsets(categories.COMPARABLE_TYPES, (hints.comparable, hints.integer,))
     _add_to_db_type_hintsets(categories.JSON_ARRAY, (hints.json_array,))
+    _add_to_db_type_hintsets(categories.ARRAY, (hints.array,))
```

Comment:
> `db/types/hintsets.py` also has to be updated, so that appropriate hints are mapped to appropriate types. Notice that we did JSON arrays before we did general arrays, so slight refactoring might be in order. 

@dmos62 is this what you are referring to?